src/HIE/Bios/Cradle.hs

```diff
@@ -381,11 +381,25 @@ cabalCradle wdir mc =
         }
     }
 
-cabalCradleDependencies :: FilePath -> IO [FilePath]
-cabalCradleDependencies rootDir = do
-    cabalFiles <- findCabalFiles rootDir
-    return $ cabalFiles ++ ["cabal.project", "cabal.project.local"]
-
+-- | @'cabalCradleDependencies' rootDir componentDir@.
+-- Compute the dependencies of the cabal cradle based
+-- on the cradle root and the component directory.
+--
+-- Directory 'componentDir' is a sub-directory where we look for
+-- package specific cradle dependencies, such as a '.cabal' file.
+--
+-- Found dependencies are relative to 'rootDir'.
+cabalCradleDependencies :: FilePath -> FilePath -> IO [FilePath]
+cabalCradleDependencies rootDir componentDir = do
+    let relFp = makeRelative rootDir componentDir
+    cabalFiles' <- findCabalFiles componentDir
+    let cabalFiles = map (relFp </>) cabalFiles'
+    return $ map normalise $ cabalFiles ++ ["cabal.project", "cabal.project.local"]
```

Comment:
Does `findCabalFiles` need to change into the `rootDir` here?

Reply:
I dont think so, `findCabalFiles` needs to look in the sub-directory of a cradle if there is one. E.g. fir hie-plugin-api, the relevant cabal file is in the sub directory. Then it returns cabal files relative to the component directory. Afterwards we need to make the dependencies relative to the root dir, otherwise our promise that dependencies are relative to the root dir cannot be upheld.